### ISSUE TITLE
Why on zero padding the dimensions increases by more than one. 

### ISSUE THREAD

**Issue by srinivasPrashanth027** [keras-team/keras#10711]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by srinivasPrashanth027** [comment]
I have used keras on mnist data set and on adding model.add(ZeroPadding2D((1,1))) the dimensions increases by more than one . Kindly refer to code below 


model=Sequential()
# 1st conv
model.add(Conv2D(96,kernel_size=(11,11),strides=(4,4),activation='relu',input_shape=(28,28,1)))
model.add(MaxPooling2D(pool_size=(3,3),strides=(2,2)))
# 2nd convl
model.add(Conv2D(256,kernel_size=(5,5),border_mode='same',activation='relu',dilation_rate=2))
model.add(MaxPooling2D(pool_size=(2,2),strides=(1,1)))

I have added zero padding layer here 
 model.add(ZeroPadding2D((1,1)))
 model.add(Conv2D(384,kernel_size=(3,3),activation='relu'))
model.add(MaxPooling2D(pool_size=(4,4)))
# when i run the code i get an error like below: 
Negative dimension size caused by subtracting 4 from 1 for 'max_pooling2d_227/MaxPool' (op: 'MaxPool') with input shapes: **_[?,3,3,256]_**

Ideally when we calculate the dimensions layer by layer 
Input= 28*28, 
1st convlayer=96 kernels, 11*11 dimension,strides=4 each so our output of this layer will have dimension of 5*5*96
On maxpooling above with strides 2 and 2*2 dimension we get 2*2*96
2nd convolution layer: input dimension for this layer is 2*2*96 and on convolution using 256 kernels of 5*5 dimension keeping padding='same', we would get 2*2*256 
Maxpool: on further applying maxpool of 2*2 as pool layer with strides as (2,2) we get 1*1*256
Now when we add one zero padding layer - model.add(ZeroPadding2D((1,1)))
we should ideally have 2*2*256 but in the error its showing the dimension at this point is 3*3*256

Help me with why on adding zeropadding the dimension is increasing by more than one ?

**Comment by tomastheod-ITI** [comment]
`ZeroPadding2D((1,1))` performs symmetric padding: +1 to top and bottom, +1 to left and right. So the size is increased by 2 in each dimension. But if you use it like this `ZeroPadding2D(padding=((1,0), (1,0)))` you control each side separately: +1 to top, +1 to left.

**Comment by Dref360** [comment]
Solved, also please use StackOverFlow, Google groups or Slack for those types of questions.

Thanks
